File: microservices/taass-spring-group-api/src/main/java/com/mfr/taass/spring/group/api/entities/Goal.java
package com.mfr.taass.spring.group.api.entities;

import java.util.Date;
import javax.persistence.Basic;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.validation.constraints.NotNull;

/**
 * @author matteo
 */
@Entity
public class Goal {

    @Id
    @GeneratedValue
    private Long id;

    @Basic
    @Column(nullable = false)
    @NotNull
    private String name;

   

    @Basic
    private String description;

    @NotNull
    private Long startDate;
    
    @NotNull
    private Long deadLine;

    @Basic
    @Column(nullable = false)
    private Long amount;

    @OneToOne
    @NotNull
    private Account account;

    @ManyToOne
    @NotNull
    private Groups groups;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }



    public Long getAmount() {
        return amount;
    }

    public void setAmount(Long amount) {
        this.amount = amount;
    }

    public Account getAccount() {
        return account;
    }

    public void setAccount(Account account) {
        this.account = account;
    }

    public Groups getGroups() {
        return groups;
    }

    public void setGroups(Groups groups) {
        this.groups = groups;
    }
    
     public Long getStartDate() {
        return startDate;
    }

    public void setStartDate(Long startDate) {
        this.startDate = startDate;
    }

    public Long getDeadLine() {
        return deadLine;
    }

    public void setDeadLine(Long deadLine) {
        this.deadLine = deadLine;
    }
    
     public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

}

File: microservices/taass-angular-frontend/src/app/types/Goal.ts
import { AccountInterface } from './Account';
import { GroupInterface } from './Group';

export interface GoalInterface
{
    id: number
    name: string
    deadLine: Date
    startDate: Date
    amountRequested:number
    amountReached?:number
    account: number    
    group:number
    description:string
    status?:string
}
export class Goal implements GoalInterface{
    id: number
    name: string
    deadLine:Date
    startDate: Date
    amount:number
    account: number   
    amountReached?:number
    amountRequested: number    
    group:number
    description:string
    status?:string

    constructor (jsonObj:GoalInterface={
        id: -1,
        name: "",
        deadLine:new Date(),
        startDate:new Date(),
        amountRequested:100,
        amountReached:0,
        account: -1  ,  
        group:-1,
        description:"",
    }){
        this.id=jsonObj.id
        this.name=jsonObj.name
        this.deadLine=jsonObj.deadLine
        this.startDate=jsonObj.startDate

        this.amountRequested=jsonObj.amountRequested
        this.amountReached=jsonObj.amountReached

        this.account=jsonObj.account
        this.group=jsonObj.group
        this.description=jsonObj.description
        }
}



